### DISCUSSION TITLE
Updating nvim-treesitter queries

### DISCUSSION THREAD

**Discussion by susliko** [tree-sitter/tree-sitter-scala#283] [General]
I think it would be nice to have some fixed process of reviewing and synchronizing changes to nvim-treesitter queries (highlights/locals/indents/etc.). 

For example:
* relevant nvim-treesitter queries are also placed in this repo (e.g. by path `queries/nvim`)
* whenever PR with changes to these queries gets merged into this repo, someone from the maintainers  (for example, the one who presses Merge) creates a PR with the same changes into nvim-treesitter

NOTE: nvim highlight queries can use different highlight groups than vanilla tree-sitter highlight queries, which we already have here

**Comment by eed3si9n** [comment]
See also
- Helix: https://github.com/helix-editor/helix/blob/master/runtime/queries/scala/highlights.scm
- Emacs: https://github.com/KaranAhlawat/scala-ts-mode/blob/main/scala-ts-mode.el

**Comment by susliko** [comment]
@ckipp01 You opinion would be very valuable as you're putting efforts to keep nvim-treesitter queries in a relevant state

**Comment by ckipp01** [comment]
>  think it would be nice to have some fixed process of reviewing and synchronizing changes to nvim-treesitter queries (highlights/locals/indents/etc.).

Firstly, I totally agree. It's not idea to have to manually sync these. However, I feel pretty strongly that the actual language tree-sitter repo (this one) should be the source of truth and other repos like nvim-treesitter should fetch from it, not maintain their own. @keynmol actually brought this up [here](https://github.com/nvim-treesitter/nvim-treesitter/pull/4055). From my understanding though not all of the integrations have the same captures so it may not always work to sync with the upstream queries and expect it work perfectly.

So I don't really have a great answer or solution. I mean we can make a check list or something that when the highlights here change we know where we they _ought_ to be updated in nvim-treesitter, helix, scala-ts-mode, but that would still require someone familiar with how they are being used in those places to probably do it.
